scipy/stats/_sensitivity_analysis.py

```diff
@@ -0,0 +1,418 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, make_dataclass
+from typing import Callable, List, TYPE_CHECKING, Tuple
+
+import numpy as np
+from scipy.stats._resampling import BootstrapResult
+
+from scipy.stats._qmc import check_random_state
+from scipy.stats import qmc, bootstrap
+from scipy.stats.sampling import NumericalInversePolynomial
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    from scipy.stats._unuran.unuran_wrapper import PINVDist
+    from scipy._lib._util import IntNumber, SeedType
+
+
+__all__ = [
+    'sobol_indices',
+    'f_ishigami'
+]
+
+
+def f_ishigami(x: npt.ArrayLike) -> np.ndarray:
+    r"""Ishigami function.
+
+    .. math::
+
+        Y(\mathbf{x}) = \sin x_1 + 7 \sin^2 x_2 + 0.1 x_3^4 \sin x_1
+
+    with :math:`\mathbf{x} \in [-\pi, \pi]^3`.
+
+    Parameters
+    ----------
+    x : array_like (n, [x1, x2, x3])
+
+    Returns
+    -------
+    f : array_like (n, 1)
+        Function evaluation.
+
+    References
+    ----------
+    .. [1] Ishigami, T. and T. Homma. "An importance quantification technique
+       in uncertainty analysis for computer models." IEEE,
+       :doi:`10.1109/ISUMA.1990.151285`, 1990.
+    """
+    x = np.atleast_2d(x)
+    f_eval = (
+        np.sin(x[:, 0])
+        + 7 * np.sin(x[:, 1])**2
+        + 0.1 * (x[:, 2]**4) * np.sin(x[:, 0])
+    )
+    return f_eval.reshape(-1, 1)
+
+
+def sample_A_B(
+    n: int,
+    dists: List[PINVDist],
+    random_state: SeedType = None
+) -> Tuple[np.ndarray,  np.ndarray]:
+    """Sample two matrices A and B.
+
+    Uses a Sobol' sequence with 2`d` columns to have 2 uncorrelated matrices.
+    This is more efficient than using 2 random draw of Sobol'.
+    See sec. 5 from [1]_.
+
+    References
+    ----------
+    .. [1] Saltelli, A., P. Annoni, I. Azzini, F. Campolongo, M. Ratto, and
+       S. Tarantola. "Variance based sensitivity analysis of model
+       output. Design and estimator for the total sensitivity index."
+       Computer Physics Communications, 181(2):259-270,
+       :doi:`10.1016/j.cpc.2009.09.018`, 2010.
+    """
+    d = len(dists)
+    A_B = qmc.Sobol(d=2*d, seed=random_state, bits=64).random(n)
+
+    A, B = A_B[:, :d], A_B[:, d:]
+
+    for d_, dist in enumerate(dists):
+        dist_rng = NumericalInversePolynomial(
+            dist, random_state=random_state
+        )
+        A[:, d_] = dist_rng.ppf(A[:, d_])
+        B[:, d_] = dist_rng.ppf(B[:, d_])
+
+    return A, B
+
+
+def sample_AB(A: np.ndarray, B: np.ndarray) -> np.ndarray:
+    """AB matrix.
+
+    AB: columns of B into A. Shape (n*d, d). e.g in 2d
+    Take A and replace 1st column with 1st column of B. You have (n, d)
+    Then A and replace 2nd column with 2nd column of B. You have (n, d)
+    Concatenate to get AB. Which means AB is (2*n, d).
+    """
+    n, d = A.shape
+    AB = np.empty((int(d*n), d))
+    for i in range(d):
+        AB[i*n:(i+1)*n, :] = np.column_stack((A[:, :i], B[:, i], A[:, i+1:]))
```

Comment:
Looks like you didn't reply to this one?

Reply:
Oops yes. Sure since I added a clear description above it's fine. I would keep it 2D (n, d) as it is now so it's easier for users. (The indice function itself is taking in the function output, not the samples, I think there was a confusion here and I think it's ok now? LMK if I should add more comments to clarify)